qubes/vm/__init__.py

```diff
@@ -547,33 +591,27 @@ def start_qdb_watch(self, loop=None):
         for path in self._qdb_watch_paths:
             self._qdb_connection_watch.watch(path)
 
-    @qubes.stateless_property
-    def klass(self):
-        """Domain class name"""
-        return type(self).__name__
-
 
 class VMProperty(qubes.property):
     """Property that is referring to a VM
 
-    :param type vmclass: class that returned VM is supposed to be instance of
+    :param type vmclass: class that returned VM is supposed to be an instance of
 
-    and all supported by :py:class:`property` with the exception of ``type`` \
-        and ``setter``
+    and all supported by :py:class:`property` except ``type`` and ``setter``
     """
 
     _none_value = ""
 
-    def __init__(self, name, vmclass=BaseVM, allow_none=False, **kwargs):
+    def __init__(self, name, vmclass=LocalVM, allow_none=False, **kwargs):
         if "type" in kwargs:
             raise TypeError(
                 "'type' keyword parameter is unsupported in {}".format(
                     self.__class__.__name__
                 )
             )
-        if not issubclass(vmclass, BaseVM):
+        if not issubclass(vmclass, LocalVM):
```

Comment:
While default to LocalVM makes sense, should it really forbid using RemoteVM as some property value?